File: 8.Best-Practice-e-trucchi-avanzati/objc/CM_HandOff/CM_HandOff/ViewController.m
//
//  ViewController.m
//  CM_HandOff
//
//  Created by WhiteTiger "sgama la rete" on 28/02/16.
//  Copyright © 2016 WhiteTiger "sgama la rete". All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (void)restoreUserActivityState:(NSUserActivity *)activity {
    if ([activity userInfo]) {
        NSString *option = [[activity userInfo] objectForKey:@"option"];
        
        UIAlertController *alert = [UIAlertController alertControllerWithTitle:@"Popup" message:[NSString stringWithFormat:@"Scelta: %@", option] preferredStyle:(UIAlertControllerStyleAlert)];
        [alert addAction:[UIAlertAction actionWithTitle:@"Close" style:(UIAlertActionStyleDestructive) handler:nil]];
        
        [self presentViewController:alert animated:YES completion:nil];
    }
    
    [super restoreUserActivityState:activity];
}

@end

File: 8.Best-Practice-e-trucchi-avanzati/swift/CM_HandOff/CM_HandOff/ViewController.swift
//
//  Copyright © 2015 WhiteTiger "sgama la rete". All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    
    override func viewDidLoad() {
        super.viewDidLoad()
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
    }
    
    override func restoreUserActivityState(activity: NSUserActivity) {
        if let local_activity = activity.userInfo {
            if let option = local_activity["option"] {
                let alert = UIAlertController(title: "Popup", message: "Scelta: \(option)", preferredStyle: .Alert)
                
                alert.addAction(UIAlertAction(title: "Close", style: .Destructive, handler: nil))
                
                self.presentViewController(alert, animated: true, completion: nil)
            }
        }
        
        super.restoreUserActivityState(activity)
    }
    
}
